src/test/unit/unittest_displayport.h

```diff
@@ -0,0 +1,163 @@
+
+static int displayPortTestGrab(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+static int displayPortTestRelease(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+static int displayPortTestClearScreen(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    memset(testDisplayPortBuffer, ' ', UNITTEST_DISPLAYPORT_BUFFER_LEN);
+    return 0;
+}
+
+static int displayPortTestDrawScreen(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+static int displayPortTestScreenSize(const displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+static int displayPortTestWrite(displayPort_t *displayPort, uint8_t x, uint8_t y, const char *s)
+{
+    UNUSED(displayPort);
+    for (unsigned int i = 0; i < strlen(s); i++) {
+        testDisplayPortBuffer[(y * UNITTEST_DISPLAYPORT_COLS) + x + i] = s[i];
+    }
+    return 0;
+}
+
+static int displayPortTestWriteChar(displayPort_t *displayPort, uint8_t x, uint8_t y, uint8_t c)
+{
+    UNUSED(displayPort);
+    testDisplayPortBuffer[(y * UNITTEST_DISPLAYPORT_COLS) + x] = c;
+    return 0;
+}
+
+static bool displayPortTestIsTransferInProgress(const displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+static int displayPortTestHeartbeat(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+static void displayPortTestResync(displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+}
+
+static uint32_t displayPortTestTxBytesFree(const displayPort_t *displayPort)
+{
+    UNUSED(displayPort);
+    return 0;
+}
+
+static const displayPortVTable_t testDisplayPortVTable = {
+    .grab = displayPortTestGrab,
+    .release = displayPortTestRelease,
+    .clearScreen = displayPortTestClearScreen,
+    .drawScreen = displayPortTestDrawScreen,
+    .screenSize = displayPortTestScreenSize,
+    .write = displayPortTestWrite,
+    .writeChar = displayPortTestWriteChar,
+    .isTransferInProgress = displayPortTestIsTransferInProgress,
+    .heartbeat = displayPortTestHeartbeat,
+    .resync = displayPortTestResync,
+    .txBytesFree = displayPortTestTxBytesFree
+};
+
+displayPort_t *displayPortTestInit(void)
+{
+    displayInit(&testDisplayPort, &testDisplayPortVTable);
+    testDisplayPort.rows = UNITTEST_DISPLAYPORT_ROWS;
+    testDisplayPort.cols = UNITTEST_DISPLAYPORT_COLS;
+    return &testDisplayPort;
+}
+
+void displayPortTestPrint(void)
+{
+    for (int i = 0; i < UNITTEST_DISPLAYPORT_BUFFER_LEN; i++) {
+        if (i > 0 && i % UNITTEST_DISPLAYPORT_COLS == 0) {
+            printf("\n");
+        }
+        printf("%c", testDisplayPortBuffer[i]);
+    }
+    printf("\n\n");
+}
+
+void displayPortTestBufferIsEmpty()
+{
+    for (size_t i = 0; i < UNITTEST_DISPLAYPORT_BUFFER_LEN; i++) {
+        EXPECT_EQ(' ', testDisplayPortBuffer[i]);
+    }
+}
+
+void displayPortTestBufferSubstring(int x, int y, const char * expectedFormat, ...)
```

Comment:
`void displayPortTestBufferSubstring(int x, int y, const char * expectedFormat, ...) __attribute__ ((format (printf, 3, 4)));` will check that arguments match format string